### ISSUE TITLE
should allow use `export = ` in `.cts` when module is esnext

### ISSUE THREAD

**Issue by bluelovers** [microsoft/TypeScript#51018]
# Bug Report

### 🕗 Version & Regression Information

- This is a crash
- This changed between versions ______ and _______
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________
- I was unable to test this on prior versions because _______

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5JA)

### 💻 Code

index.cts

```ts
import { globToTree } from './index';

export = globToTree
```

index.ts
```ts
export function globToTree(): any {}
Object.defineProperty(globToTree, "__esModule", { value: true });
Object.defineProperty(globToTree, "globToTree", { value: globToTree });
Object.defineProperty(globToTree, "default", { value: globToTree });
export default globToTree
```

```json
{
  "compilerOptions": {
    "module": "esnext",
    "target": "es2019",
    "jsx": "preserve",
    "moduleResolution": "node",
    "allowJs": false,
    "allowUmdGlobalAccess": false,
    "sourceMap": true,
    "inlineSourceMap": false,
    "inlineSources": true,
    "declaration": false,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "strict": true,
    "strictBindCallApply": true,
    "strictNullChecks": false,
    "strictFunctionTypes": true,
    "noImplicitThis": true,
    "noImplicitOverride": true,
    "noUncheckedIndexedAccess": true,
    "noPropertyAccessFromIndexSignature": true,
    "preserveConstEnums": true,
    "forceConsistentCasingInFileNames": true,
    "incremental": true,
    "noErrorTruncation": true,
    "removeComments": false,
    "traceResolution": false,
    "newLine": "lf",
    "esModuleInterop": true,
    "importHelpers": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "skipLibCheck": true,
    "noEmit": true
  }
}
```

### 🙁 Actual behavior

show error

![image](https://user-images.githubusercontent.com/167966/193362563-7b3d59f1-133b-4ab5-a599-0f236de7be7c.png)

or use `@ts-ignore` avoid it

```ts
import { globToTree } from './index';

// @ts-ignore
export = globToTree
```

### 🙂 Expected behavior

no need `@ts-ignore`

**Comment by RyanCavanaugh** [comment]
What code is supposed to be generated in this case?

**Comment by bluelovers** [comment]
input
https://github.com/bluelovers/ws-glob/blob/master/packages/glob-tree-list/src/index.cts
https://github.com/bluelovers/ws-glob/blob/master/packages/glob-tree-list/src/index.ts


output
https://github.com/bluelovers/ws-glob/blob/master/packages/glob-tree-list/dist/index.cjs.development.cjs
https://github.com/bluelovers/ws-glob/blob/master/packages/glob-tree-list/dist/index.esm.mjs

**Comment by RyanCavanaugh** [comment]
https://github.com/bluelovers/ws-glob/blob/master/packages/glob-tree-list/dist/index.esm.mjs - this is just a `default` export, like it's telling you to do. `export = (a function` doesn't make sense in an ES Module output format. What's the motivation for not writing `export default globToTree` ?

**Comment by fatcerberus** [comment]
@RyanCavanaugh I guess they’re expecting the `.cts` file to compile to CommonJS instead of ESM?

**Comment by andrewbranch** [comment]
This was implemented as a new feature in 5.6 (https://github.com/microsoft/TypeScript/pull/58825)
